narwhals/dataframe.py

```diff
@@ -628,6 +628,42 @@ def columns(self) -> list[str]:
         """
         return super().columns
 
+    def rows(
+        self, *, named: bool = False
+    ) -> list[tuple[Any, ...]] | list[dict[str, Any]]:
+        """
+        Returns all data in the DataFrame as a list of rows of python-native values.
+
+        By default, each row is returned as a tuple of values given in the same order
+        as the frame columns. Setting named=True will return rows of dictionaries instead.
+
+        Examples:
+            >>> import pandas as pd
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> df = {"foo": [1, 2, 3], "bar": [6.0, 7.0, 8.0], "ham": ["a", "b", "c"]}
+            >>> df_pd = pd.DataFrame(df)
+            >>> df_pl = pl.DataFrame(df)
+
+            We define a library agnostic function:
+
+            >>> def func(df_any, named):
```

Comment:
nitpick, but this is a "boolean trap" - can we make `named` keyword-only for the example please?

i.e.
```
func(df_any, *, named):
```

then, when you call it - `func(df_pd, named=False)`, etc

Reply:
Addressed.